macros/get_metric_sql.sql

```diff
@@ -30,20 +34,14 @@ VALIDATION ROUND ONE - THE MACRO LEVEL!
     {%- do exceptions.raise_compiler_error("No date grain provided") %}
 {%- endif %}
 
-{% for metric in metric_list %}
-    {% if metric.type != "expression" and metric.metrics | length > 0 %}
-        {%- do exceptions.raise_compiler_error("The metric " ~ metric.name ~ " was not an expression and dependent on another metric. This is not currently supported - if this metric depends on another metric, please change the type to expression.") %}
-    {%- endif %}
-{% endfor %}
+{% do metrics.validate_grain(grain,metric_tree['full_set'])%}
+
+{% do metrics.validate_expression_metrics(metric_tree['full_set'])%}
 
 {# ############
 LETS SET SOME VARIABLES!
 ############ #}
 
-{# We are creating the metric tree here - this includes all the leafs (first level parents)
-, the expression metrics, and the full combination of them both #}
-{%- set metric_tree = metrics.get_metric_tree(metric_list) %}
```

Comment:
I just move this to the top because we reference the metric tree in the earlier validation now.